doc/nrf/gs_modifying.rst

```diff
@@ -151,8 +221,39 @@ To locate a specific configuration option, use the filter (:guilabel:`Jump to` i
 The documentation for each :ref:`configuration option <configuration_options>` also lists the menu path where the option can be found.
 
 .. important::
-   All changes to the :file:`.config` file are lost when you clean your build directory.
-   You can save it to another location, but you must then manually copy it back to your build directory.
+   All changes to the :file:`.config` file are lost when you clean your build directory or you rebuild the project with the ``pristine`` target.
+   You can save the file to another location, but you must then manually copy it back to your build directory.
+   See Zephyr's :ref:`zephyr:application_rebuild` for more information.
+
+Changing the configuration permanently
+======================================
+
+To configure your application and maintain the configuration when you clean the build directory, you need to specify the configuration in one of the permanent configuration files.
+In most cases, this will be either the default :file:`prj.conf` file of the application or one of the configuration file overlays.
+In these files, you can specify different values for configuration options that are defined by a library or board, and you can add configuration options that are specific to your application.
+
+See :ref:`zephyr:setting_configuration_values` in the Zephyr documentation for information on how to edit the :file:`prj.conf` file.
+
+.. tip::
+   When modifying configuration, especially when disabling one of the previously enabled options, check if the options required by the option you are disabling are still needed.
```

Comment:
Answering on behalf of Pawel as he is unavailable.

Alternative suggestion:

Modifying the temporary configuration can result in an invalid or bloated configuration due to default values already having been calculated based on the previous configuration. It is therefore good practice to modify :file:`prj.conf` instead.

Reply:
That's what Pawel meant, @SebastianBoe , when he was explaining that to me, thanks.

Reply:
> Modifying the temporary configuration can result in an invalid .... configuration

if the result of custom changes to the project using for example menuconfig results in an invalid configuration, then I believe we have a dependency issue in our Kconfig which should be fixed.

`menuconfig` and friends are still valid tools to try out temporary changes before deciding if they should be applied permanently.

Reply:
Aye, but we do have dependency issues. Which is what this tip is about, a safe way to change configurations without being affected by dependency issues.

I agree with Pawel that it is good practice to not use menuconfig to change options. There are exceptions to the rule, but it is a good tip.

If you wish we could rephrase it to say "to make changes in a way safe from dependency issues", or similar.

It's the same as building. I wouldn't do a release with an incremental build. To be safe, I would do a clean build, even though incremental builds are supposed to work.

Reply:
> but we do have dependency issues.

where ?
I have tried to followup everytime I've discovered those, like this cleanup but also parts related to bluetooth:
https://github.com/zephyrproject-rtos/zephyr/pull/34594
https://github.com/nrfconnect/sdk-zephyr/pull/510
https://github.com/nrfconnect/sdk-nrf/pull/4402

so I would personally think we should treat this as errors and fix them.

Also, we don't state in general, `code might have undiscovered bugs`.

If you know for a fact that there are dedicated places where we dependency issues, please list them, and we can fix those.

Reply:
I see them every time I read the code, my impression is that of the thousands of configs, there are hundreds that can not be modified through menuconfig without creating a build that is different from the one created if it were modified in prj.conf.

One of these hundreds is: https://projecttools.nordicsemi.no/jira/browse/NCSDK-11195?filter=-1&jql=reporter%20%3D%20anvi%20%20order%20by%20updated%20DESC

I don't have time to go looking for more.

I understand you have a different impression, that there are only a few dozen perhaps, of the thousands of configs.

In any case, as code owner, you may decide what kind of usage we should be encouraging. I thought it was 
a valid tip for a common problem, but if you see otherwise then removing this tip OK with me.

Reply:
> there are hundreds that can not be modified through menuconfig without creating a build that is different from the one created if it were modified in prj.conf.

Yes, the two builds may be different based on whether `menuconfig` was used or the `prj.conf` was edited directly.
But even a bloated configuration should be valid.
That is why this sentence is true and should be documented as you pointed out:
> Modifying the temporary configuration can result in a bloated configuration

> I understand you have a different impression, that there are only a few dozen perhaps, of the thousands of configs.

We probably have thousands of places that can result in a bloated configuration, but we should not have places resulting in an invalid configuration, cause if we have then such an invalid configuration could even be created by hand in a `prj.conf` directly.

> One of these hundreds is: https://projecttools.nordicsemi.no/jira/browse/NCSDK-11195?filter=-1&jql=reporter%20%3D%20anvi%20%20order%20by%20updated%20DESC

and this is exactly filed as a bug, and should be treated as such.
That one should probably even be marked as a known issue.

What about writing:
> Reconfiguring through `menuconfig` only change the specific setting and will not adjust defaults for other visible and configured symbols, see [Stuck symbols in menuconfig and guiconfig](http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-6121/zephyr/guides/build/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig). This may result in a bloated configuration compared to changing the setting directly in `prj.conf`.

Reply:
+1 to your re-write.

Reply:
s/only change/only changes/